src/TMap.cpp

```diff
@@ -3405,21 +3405,34 @@ bool TMap::incrementJsonProgressDialog(const bool isExportNotImport, const bool
     return mpProgressDialog->wasCanceled();
 }
 
+/**
+ * Update the the 2D and 3D map visually.
+ *
+ * It ensures debouncing internally to ensure that bulk calls are efficient.
+ */
 void TMap::update()
 {
+    static bool debounce;
+    if (!debounce) {
+        debounce = true;
+        QTimer::singleShot(0, this, [this]() {
+            debounce = false;
+
 #if defined(INCLUDE_3DMAPPER)
-    if (mpM) {
-        mpM->update();
-    }
+            if (mpM) {
+                mpM->update();
+            }
 #endif
-    if (mpMapper) {
-        mpMapper->checkBox_showRoomNames->setVisible(getRoomNamesPresent());
-        mpMapper->checkBox_showRoomNames->setChecked(getRoomNamesShown());
+            if (mpMapper) {
+                mpMapper->checkBox_showRoomNames->setVisible(getRoomNamesPresent());
+                mpMapper->checkBox_showRoomNames->setChecked(getRoomNamesShown());
 
-        if (mpMapper->mp2dMap) {
-            mpMapper->mp2dMap->mNewMoveAction = true;
-            mpMapper->mp2dMap->update();
-        }
+                if (mpMapper->mp2dMap) {
+                    mpMapper->mp2dMap->mNewMoveAction = true;
```

Comment:
If you can work out the program logic behind the usage of this flag you might be better placed (than me) to say whether it should be done in every case (or not) - since I didn't grok the original usage I can't say whether this is required in here - or should only have been done in some cases back when it was in the `TLuaInterpreter` class code.

Reply:
I am not sure either, and getting into this would be outside of the scope of this PR :/